src/app/utils/date.ts

```diff
@@ -0,0 +1,21 @@
+import moment from 'moment-timezone'
+
+export const isMalformedDate = (date?: string): Boolean => {
+  return Boolean(date) && !moment(date, 'YYYY-MM-DD', true).isValid()
```

Comment:
Just to check, why is Boolean(date) necessary - wouldn't date suffice?

Reply:
just to enforce that the return type is `boolean`. for instance,
```
undefined && false = undefined
Boolean(undefined) && false = false
```